rest/admin_api_test.go

```diff
@@ -2966,3 +2976,142 @@ func TestObtainUserChannelsForDeletedRoleCasFail(t *testing.T) {
+		expectWarn          bool
+	}{
+		{
+			name:                "Over max default channel length",
+			warnThresholdLength: base.DefaultWarnThresholdChannelNameSize,
+			channelLength:       int(base.DefaultWarnThresholdChannelNameSize) + 1,
+			expectWarn:          true,
+		},
+		{
+			name:                "Equal to max default channel length",
+			warnThresholdLength: base.DefaultWarnThresholdChannelNameSize,
+			channelLength:       int(base.DefaultWarnThresholdChannelNameSize),
+			expectWarn:          false,
+		},
+		{
+			name:                "Under max default channel length",
+			warnThresholdLength: base.DefaultWarnThresholdChannelNameSize,
+			channelLength:       int(base.DefaultWarnThresholdChannelNameSize) - 1,
+			expectWarn:          false,
+		},
+		{
+			name:                "Over max configured channel length",
+			warnThresholdLength: 500,
+			channelLength:       501,
+			expectWarn:          true,
+		},
+		{
+			name:                "Equal to max configured channel length",
+			warnThresholdLength: 500,
+			channelLength:       500,
+			expectWarn:          false,
+		},
+		{
+			name:                "Under max configured channel length",
+			warnThresholdLength: 500,
+			channelLength:       499,
+			expectWarn:          false,
+		},
+	}
+
+	for _, test := range testCases {
+		t.Run(test.name, func(t *testing.T) {
+			var thresholdConfig db.WarningThresholds
+			// If threshold is not default then configure it
+			if test.warnThresholdLength != base.DefaultWarnThresholdChannelNameSize {
+				thresholdConfig = db.WarningThresholds{ChannelNameSize: &test.warnThresholdLength}
+			}
+			rt = NewRestTester(t, &RestTesterConfig{
+				SyncFn: syncFn,
+				DatabaseConfig: &DbConfig{
+					Unsupported: db.UnsupportedOptions{
+						WarningThresholds: thresholdConfig,
+					},
+				},
+			})
+			defer rt.Close()
+
+			chanNameWarnCountBefore := rt.ServerContext().Database("db").DbStats.Database().WarnChannelNameSizeCount.Val
+			tr := rt.SendAdminRequest("PUT", fmt.Sprintf("/db/doc%v", test.channelLength), fmt.Sprintf("{\"chan\":\"%s\"}", strings.Repeat("A", test.channelLength)))
+			assertStatus(t, tr, http.StatusCreated)
+			chanNameWarnCountAfter := rt.ServerContext().Database("db").DbStats.Database().WarnChannelNameSizeCount.Val
+			if test.expectWarn {
+				assert.Equal(t, chanNameWarnCountBefore+1, chanNameWarnCountAfter)
+			} else {
+				assert.Equal(t, chanNameWarnCountBefore, chanNameWarnCountAfter)
+			}
+		})
+	}
+}
+func TestChannelNameSizeWarningUpdateExistingDoc(t *testing.T) {
+	syncFn := "function sync(doc, oldDoc) { channel(doc.chan); }"
+	rt := NewRestTester(t, &RestTesterConfig{
+		SyncFn: syncFn,
+	})
+	defer rt.Close()
+
+	// Update doc - should warn
+	channelLength := int(base.DefaultWarnThresholdChannelNameSize) + 5
+	t.Run("Update doc without changing channel", func(t *testing.T) {
+		tr := rt.SendAdminRequest("PUT", "/db/replace", fmt.Sprintf("{\"chan\":\"%s\"}", strings.Repeat("B", channelLength))) // init doc
```

Comment:
(nitpick) With it being the same channel name can we assign a variable here for the channel?